base/readiness.go

```diff
@@ -0,0 +1,207 @@
+package base
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	log "github.com/iter8-tools/iter8/base/log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	// ReadinessTaskName is the task name
+	ReadinessTaskName = "k8s-object-ready"
+
+	// defaultTimeout is default timeout for readiness command
+	defaultTimeout = "10s"
+)
+
+// ReadinessInputs identifies the K8s object to test for existence and
+// the (optional) condition that should be tested (succeeds if true).
+type readinessInputs struct {
```

Comment:
The task requires:
```go
Group: string
Version string
Resource: string
```

We are taking as inputs,
```go
APIVersion: string
Resource: string
```
and leaving out the group. 

And we're using a `gvr` function to parse out the `APIVersion` into `version` and `group`.

***

Could we please use the former, much more direct fields in the task? i.e., 
```go
Group: string
Version string
Resource: string
```

This will obviously simplify the task implementation. For example, it would get rid of the need for `gvr` function, as you will straight away use what is passed in as inputs to the task.

*** 
Perhaps less obvious, it would also simplify the task template definition and invocation. In particular, it would allow you to craft a template like ... 
```go
{{- define "task.ready" }}
- task: k8s-object-ready
  with:
{{ toYaml . | indent 4 }}
{{- end }}
```

and invoke it like this.

```go
{{- if .Values.service.name }}
{{ include "task.ready" .Values.service }}
{{- end }}
```

and 

```go
{{- if .Values.deploy.name }}
{{ include "task.ready" .Values.deploy }}
{{- end }}
```

Reply:
I made this change but I am not sure it does either of these things as cleanly as suggested. I may not understand the template approach.  But this is a separate discussion.